aot-std-optimizers/src/test/groovy/io/micronaut/aot/std/sourcegen/YamlPropertySourceGeneratorTest.groovy

```diff
@@ -56,35 +57,31 @@ public class Test_configStaticPropertySource extends MapPropertySource {
   }
 
   private static List list2() {
-    List result = new ArrayList<>();
-    result.add("isAnonymous()");
-    return result;
+    return Collections.singletonList("isAnonymous()");
   }
 
   private static Map map1() {
-    Map result = new LinkedHashMap<>();
+    Map result = new LinkedHashMap<>(3);
     result.put("pattern", "/api/v1/subscriber/confirm");
     result.put("http-method", "PATCH");
     result.put("access", list2());
     return result;
   }
 
   private static List list4() {
-    List result = new ArrayList<>();
-    result.add("isAnonymous()");
-    return result;
+    return Collections.singletonList("isAnonymous()");
   }
 
   private static Map map3() {
-    Map result = new LinkedHashMap<>();
+    Map result = new LinkedHashMap<>(3);
     result.put("pattern", "/api/v1/subscriber/count");
     result.put("http-method", "GET");
     result.put("access", list4());
     return result;
   }
 
   private static List list0() {
-    List result = new ArrayList<>();
+    List result = new ArrayList<>(2);
```

Comment:
One other nice optimizations would be to use `List.of` on JDK 11+ :)

Reply:
Yes, I wish I could do this now... but for now I decided to keep it simple stupid :) With Micronaut 4 we can do this.